optuna/integration/mlflow.py

```diff
@@ -239,14 +247,37 @@ def _set_tags(self, trial: optuna.trial.FrozenTrial, study: optuna.study.Study)
         # This sets the tags for MLflow.
         mlflow.set_tags(tags)
 
-    def _log_metric(self, value: Union[float, None]) -> None:
-        """Log the trial result as metric to MLflow.
+    def _log_metrics(self, values: List[Union[float, None]]) -> None:
+        """Log the trial results as metrics to MLflow.
 
         Args:
-            value: Result of trial.
+            values: Results of a trial.
         """
-        trial_value = value if value is not None else float("nan")
-        mlflow.log_metric(self._metric_name, trial_value)
+
+        if isinstance(self._metric_name, list):
+            if len(self._metric_name) != len(values):
+                raise RuntimeError(
```

Comment:
`self._metric_name` is the correct type (i.e., list) and it is specified by users, so I think we can use more specific error than [`RuntimeError`](https://docs.python.org/3/library/exceptions.html#RuntimeError), such as [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError).

```suggestion
                raise ValueError(
```

Reply:
Note: also TODO in `WeightsAndBiasesCallback`

Reply:
Addressed in ec628b84738cd8b9a84ad9f1bf7e5636aea8f718.

Reply:
Thanks!